libcontainer/init_linux.go

```diff
@@ -492,7 +492,13 @@ func signalAllProcesses(m cgroups.Manager, s os.Signal) error {
 			logrus.Warn(err)
 		}
 	}
-	if err := m.Freeze(configs.Thawed); err != nil {
+
+	if s != unix.SIGKILL {
```

Comment:
Why do we need this logic here?  I feel like branching on signals will cause some issues for users when the thaw works differently.

Reply:
We only want to recursive thaw on sigkill because that's the scenario when we have processes lying around after the parent process has exited.  For example, if we have a nested container that is frozen because it's paused, I believe the desired behavior is for it to remain frozen/paused.

Reply:
Note that below, there's similar logic that special-cases SIGKILL. This here is special-cased for the same reasons. In the SIGKILL case, runC is expecting everything everything to die (and the code below will wait for each process to die, which can hang forever without this patch).

For every other signal, runC is not expecting every process to die, so there's no problem if container processes remain frozen instead of dying.